### PR TITLE
Initial upload of company-documents project

### PR DESCRIPTION
**Pull request by amoriartyCH**
Uploading a shell project for the Alternate paper output project.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+.PHONY: clean
+clean:
+	mvn clean
+	rm -f ./$(artifact_name).jar
+	rm -f ./$(artifact_name)-*.zip
+	rm -rf ./build-*
+
+.PHONY: build
+build:
+	mvn install
```
Comment (divyaparameswaran) [inline Makefile]: Should this be mvn compile ? https://github.com/companieshouse/styleguides/blob/master/Makefiles/java_makefiles.md

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+.PHONY: all
+all: build
+
+.PHONY: clean
+clean:
+	mvn clean
+	rm -f ./$(artifact_name).jar
+	rm -f ./$(artifact_name)-*.zip
+	rm -rf ./build-*
```
Comment (divyaparameswaran) [inline Makefile:11]: Can you add `rm -f ./build.log`  as well please

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	mvn install
+
+.PHONY: test
+test: test-unit test-integration
+
+.PHONY: test-unit
+test-unit: clean
+	mvn test
+
+.PHONY: test-integration
```
Comment (divyaparameswaran) [inline Makefile]: No need for these targets just yet . We can add them later when we add integration tests .

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+test-integration: clean
+	mvn -Dtest=*IntegrationTest test
+
+.PHONY: test-contract-provider
```
Comment (divyaparameswaran) [inline Makefile]: Same as above.

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+test-contract-provider: clean
+	mvn -Dtest=*ProviderContractTest test
+
+.PHONY: test-contract-consumer
```
Comment (divyaparameswaran) [inline Makefile]: Same as above .

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,127 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+		<groupId>uk.gov.companieshouse</groupId>
+		<artifactId>companies-house-spring-boot-parent</artifactId>
+		<version>1.2.0</version>
```
Comment (divyaparameswaran) [inline pom.xml]: Are we happy with this version ? Is there a possibility to use the latest version - https://github.com/companieshouse/spring-boot-parent-pom/releases

File: src/main/java/uk/gov/companieshouse/web/companydocuments/controller/HealthCheckController.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+/**
+ * Returns HTTP OK response to indicate a healthy service is running
+ */
+@RequestMapping("/company-documents/health-check")
```
Comment (divyaparameswaran) [inline src/main/java/uk/gov/companieshouse/web/companydocuments/controller/HealthCheckController.java]: The endpoint should be as per  ->  https://github.com/companieshouse/styleguides/blob/master/health_check.md

File: pom.xml [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+	</parent>
+	<artifactId>company-documents.web.ch.gov.uk</artifactId>
+	<version>unversioned</version>
+	<name>company-documents.web.ch.gov.uk</name>
+	<description>Web application for handling the retrieval of company documents</description>
+
+	<properties>
+		<java.version>1.8</java.version>
+		<org.mapstruct.version>1.3.0.Final</org.mapstruct.version>
+		<jacoco-maven-plugin.version>0.7.7.201606060606</jacoco-maven-plugin.version>
```
Comment (divyaparameswaran) [inline pom.xml]: This can be inherited from parent .

File: pom.xml [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+				<executions>
+					<execution>
+						<goals>
+							<goal>repackage</goal>
+						</goals>
+					</execution>
+				</executions>
+			</plugin>
+
+			<plugin>
```
Comment (divyaparameswaran) [inline pom.xml]: Just this is enough 
`<plugin>
 <groupId>org.jacoco</groupId>
<artifactId>jacoco-maven-plugin</artifactId>
</plugin>`

File: pom.xml [hunk 10]
```diff
@@ -0,0 +1,127 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (divyaparameswaran) [inline pom.xml:1]: Indentation in the pom file is off . Can you use spaces instead of tabs please

File: routes.yaml [hunk 11]
```diff
@@ -0,0 +1,5 @@
+app_name: company-documents.web.ch.gov.uk
+group: web
+weight: 991
+routes:
+   1: ^/company-documents/health-check
```
Comment (divyaparameswaran) [inline routes.yaml]: Should this be `/healthcheck/* ?

File: src/main/java/uk/gov/companieshouse/web/companydocuments/CompanyDocumentsWebApplication.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.companieshouse.web.companydocuments;
```
Comment (divyaparameswaran) [inline src/main/java/uk/gov/companieshouse/web/companydocuments/CompanyDocumentsWebApplication.java:1]: Replace tabs with spaces

File: src/main/java/uk/gov/companieshouse/web/companydocuments/controller/HealthCheckController.java [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+/**
+ * Returns HTTP OK response to indicate a healthy service is running
+ */
+@RequestMapping("/company-documents/healthcheck")
```
Comment (divyaparameswaran) [inline src/main/java/uk/gov/companieshouse/web/companydocuments/controller/HealthCheckController.java]: The url may need to be changed .

**Review by divyaparameswaran — CHANGES_REQUESTED** [review]
Changes needed please

File: src/main/resources/application.yaml [hunk 14]
```diff
@@ -0,0 +1,10 @@
+server:
```
Comment (divyaparameswaran) [inline src/main/resources/application.yaml:1]: Is adding healthcheck MVP ?

File: src/test/java/uk/gov/companieshouse/web/companydocuments/controller/HealthCheckControllerIntegrationTest.java [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
```
Comment (divyaparameswaran) [inline src/test/java/uk/gov/companieshouse/web/companydocuments/controller/HealthCheckControllerIntegrationTest.java]: Avoid * imports please

File: pom.xml [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+                <exclusion>
+                    <groupId>org.junit.vintage</groupId>
+                    <artifactId>junit-vintage-engine</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>${junit-jupiter-engine-version}</version>
```
Comment (ch-tstrickland) [inline pom.xml]: I don't think that this is needed, as the property is provided by companies-house-spring-boot-parent. In my copy of Eclipse this leads to an error that points out that we are overriding an already provided version (5.5.2) with our version (5.2.0). We should probably remove this and its counterpart on line 18.

File: pom.xml [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-junit-jupiter</artifactId>
+            <version>${mockito-junit-jupiter-version}</version>
```
Comment (ch-tstrickland) [inline pom.xml]: I don't think that this is needed, as the property is provided by companies-house-spring-boot-parent. In my copy of Eclipse this leads to an error that points out that we are overriding an already provided version (3.1.0) with our version (2.18.0). We should probably remove this and its counterpart on line 19.

File: pom.xml [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+
+            <plugin>
+                <groupId>org.jacoco</groupId>
+                <artifactId>jacoco-maven-plugin</artifactId>
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+                <version>${maven-compiler-plugin.version}</version>
```
Comment (ch-tstrickland) [inline pom.xml]: Not defined in file. Could delete this line safely, as it is defined elsewhere (according to Eclipse).

**Review by ch-tstrickland — COMMENTED** [review]
Looks good to me, except for some remaining warnings in the pom.xml and some tabs in CompanyDocumentsWeApplicationTests.

File: pom.xml (hunk 18 above)
Comment (ch-tstrickland) [reply pom.xml]: Why resolved? Not deleted.

File: Makefile [hunk 19]
```diff
@@ -0,0 +1,48 @@
+artifact_name       := company-documents.web.ch.gov.uk
```
Comment (ch-tstrickland) [inline Makefile:1]: There is no test target in this file.

Comment (amoriartyCH) [reply Makefile:1]: Mentioned in slack PM, removed from earlier comment. Closing :)

**Review by divyaparameswaran — APPROVED** [review]
happy with the changes

**Pull request merged**